### DISCUSSION TITLE
`Annotated` dependencies are interpreted incorrectly when using `PEP 695`-style type alias.

### DISCUSSION THREAD

**Discussion by Rockmizu** [fastapi/fastapi#10662] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from __future__ import annotations

from typing import Annotated, TypeAlias

from fastapi import Depends, FastAPI

app = FastAPI()


# dependency
async def some_value() -> int:
    return 123

# This works.
DependedValue: TypeAlias = Annotated[int, Depends(some_value)]

# This won't work.
# type DependedValue = Annotated[int, Depends(some_value)]


@app.get('/')
async def get_with_dep(value: DependedValue) -> str:
    print(f'{type(value) = !r}')
    print(f'{value = !r}')

    assert isinstance(value, int), '`value` should be an integer.'
    assert value == 123, '`value` should be 123.'

    return f'value: {value}'
```

### Description

When using the method in line 15:
```python
DependedValue: TypeAlias = Annotated[int, Depends(some_value)]
```
to declare a type alias, FastAPI correctly interprets it and fills the `value` with `123` for `get_with_dep`.

However, when using the new type alias syntax introduced in Python 3.12 (line 18):
```python
type DependedValue = Annotated[int, Depends(some_value)]
```
FastAPI interprets it as a URL query parameter and responds with HTTP 422 Unprocessable Entity when accessing `/`.

## Expected Response

HTTP 200 OK

```
"value: 123"
```

## Actual Response

HTTP 422 Unprocessable Entity

```
{"detail":[{"type":"missing","loc":["query","value"],"msg":"Field required","input":null,"url":"https://errors.pydantic.dev/2.5/v/missing"}]}
```

### Operating System

Windows

### FastAPI Version

0.104.1

### Pydantic Version

2.5.1

### Python Version

Python 3.12.0

**Comment by 9en9i** [comment]
I also encountered this problem 😥
Apparently the problem is that when using type the type becomes TypeAliasType.

**Reply by Rockmizu** [reply]
Yes, fortunately, it's possible to extract the aliased type from `TypeAliasType` both in static analysis and at runtime. We can hope that when FastAPI officially supports 3.12, this bug will be fixed.

**Comment by Kludex** [comment, marked as answer]
I've created https://github.com/tiangolo/fastapi/issues/10719 to track this.
